### DISCUSSION TITLE
Redshift

### DISCUSSION THREAD

**Discussion by 0323pin** [leftwm/leftwm#898] [General]
Hi,

I used to start `redshift` from `.xinitrc` but, today I thought, why not getting that into my theme? After all `picom` starts from there.
Here's the relevant snip from the `up` script, `picom` included for reference.

```
# Start picom
if [ -x "$(command -v picom)" ]; then
  picom --config $SCRIPTPATH/picom.conf &> /dev/null &
fi

# Start redshift
if [ -x"$(command -v redshift)" ]; then
  redshift -c $SCRIPTPATH/redshift.conf &> /dev/null &
fi
```

Now, this works fine but, if I do a SoftReload, I get two instances of `redshift` running and the screen brightness gets adjusted every second, giving a _blinking effect_.

Just to be clear, here's the `down` script as well,

```
#!/bin/sh

SCRIPTPATH="$( cd "$(dirname "$0")" || return ; pwd -P )"

leftwm-command "UnloadTheme"

# Kill all started applications
pkill lemonbar
pkill feh
pkill picom
pkill redshift
```

Anyone using `redshift` that has some thoughts?

```
~ > leftwm check
:: LeftWM version: 0.4.0
:: LeftWM git hash: 85f832ac4b
:: Loading configuration . . .
    -> Configuration loaded OK
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK (has XDG_RUNTIME_DIR)
:: Checking theme . . .
    -> Theme OK
```

TIA

**Comment by UlisesAlexanderAM** [comment]
I use it too, but I start it differently. 

`redshift -PO 3600`

I had a similar problem in xmonad so reading the documentation, I came across the `P` flag. The redshift man page says the following:

`-P     Reset existing gamma ramps before applying new color effect.`

My hypothesis is that if you pass that flag to your current command in the `up` script, it should work.

**Comment by 0323pin** [comment]
@UlisesAlexanderAM thanks! I'll try that. Although, I might look if it's possible to have a connection to `dbus` and avoid the initiation of parallel processes.

**Comment by 0323pin** [comment]
:( Apparently, not possible, https://github.com/jonls/redshift/issues/218

Maybe it could be killed when a SoftReload is triggered 🤔

**Reply by VuiMuich** [reply]
Not a really nice solution, but when you check in the up-script if there is already a `redshift` process? E.g. with `procs redshift --no-header --only PID`

**Reply by 0323pin** [reply]
Yeah, I was thinking something like:
```
# Start redshift
killall -q redshift
if [ -x"$(command -v redshift)" ]; then
  redshift -c $SCRIPTPATH/redshift.conf &> /dev/null &
fi
```

**Comment by 0323pin** [comment]
Ok, thank you both for the hints. This works 😄 

```
# Start redshift
# Avoid starting more than one process when calling SoftReload.
# Kill redshift in case it's already running.
pkill redshift
if [ -x"$(command -v redshift)" ]; then
  redshift -P -c $SCRIPTPATH/redshift.conf &> /dev/null &
fi
```
